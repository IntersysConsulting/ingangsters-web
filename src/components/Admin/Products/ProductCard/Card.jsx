import React from "react";
import { Link } from "react-router-dom";
import "./Card.css";

const Card = ({ name, stock, image, path, available, reload }) => {
  var stocklevel;
  if (stock >= 35) {
    stocklevel = "high";
  } else if (stock >= 20) {
    stocklevel = "medium";
  } else {
    stocklevel = "low";
  }
  const classUnavailable = available ? "" : "unavailable";
  return (
    <Link
      className={"productCard " + classUnavailable}
      to={{ pathname: path, reload: reload }}
    >
      <div className="productImage">
        <img src={image} alt={name} />
      </div>
      <div className="productData">
        <p className="productName">{name}</p>
        <p className="productStock">
          Stock: <span className={stocklevel}>{stock}</span>
        </p>
      </div>
    </Link>
  );
};

export default Card;
