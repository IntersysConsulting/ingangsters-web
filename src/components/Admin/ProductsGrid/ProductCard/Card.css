.productCard {
  width: 100%;
  max-width: 400px;
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 120px;
  background-color: #ffffff;
  margin-left: 5px;
  margin-bottom: 5px;
}

.productCard > img {
  width: 50px;
  height: 50px;
}
.productCard > .productData {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}
.productCard > .productData > .productName {
  font-size: 20px;
  font-weight: bold;
  text-align: center;
}

.productCard > .productData > .productStock {
  font-size: 14px;
}

.productCard > .productData > .productStock > span.high {
  color: var(--green);
}

.productCard > .productData > .productStock > span.medium {
  color: var(--orange);
}

.productCard > .productData > .productStock > span.low {
  color: var(--red);
}
