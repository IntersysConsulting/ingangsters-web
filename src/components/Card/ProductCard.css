.product-card {
  border: 1px solid #b4d1e7;
  border-radius: 30px;
}
