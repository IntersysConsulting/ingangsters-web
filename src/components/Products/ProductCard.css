@import "../../css/colors.css";

.product-card {
  border: 2px solid #b4d1e7;
  border-radius: 30px;
  background-color: var(--white);
}

.highlight:hover {
  -webkit-box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
  -moz-box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
}

.vertical-center {
  min-height: 100%;
  display: flex;
  align-items: center;
}

.col-container {
  display: table;
  width: 100%;
}

.content-center {
  text-align: center !important;
  height: 170px;
  display: flex !important;
  flex-direction: column;
  justify-content: space-evenly;
}

.product-title {
  border-bottom: 2px solid #b4d1e7;
  min-height: 80px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.line-clamp {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.price {
  color: var(--light-blue);
}

.title-link {
  text-decoration: none !important;
  color: var(--light-blue);
}
