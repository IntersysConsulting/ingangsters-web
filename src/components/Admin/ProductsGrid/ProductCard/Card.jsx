import React from "react";
import Col from "react-bootstrap/Col";
import "./Card.css";

const Card = ({ productName, productStock, productImage, onClick }) => {
  var stocklevel;
  if (productStock >= 35) {
    stocklevel = "high";
  } else if (productStock >= 20) {
    stocklevel = "medium";
  } else {
    stocklevel = "low";
  }
  return (
    <Col className="productCard" onClick={onClick}>
      <img src={productImage} alt={productName} />
      <div className="productData">
        <p className="productName">{productName}</p>
        <p className="productStock">
          Stock: <span className={stocklevel}>{productStock}</span>
        </p>
      </div>
    </Col>
  );
};

export default Card;
