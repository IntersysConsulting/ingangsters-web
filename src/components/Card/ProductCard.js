import React, { useEffect } from "react";
import { FaCartPlus } from "react-icons/fa";
import "./ProductCard.css";
import { connect } from "react-redux";
import { getProducts } from "../../actions/creators/products";

const ProductCard = ({ getProducts, product }) => {
  useEffect(() => {
    getProducts();
  }, [getProducts]);

  return (
    <div className="container-fluid">
      <div className="row m-1 product-card">
        <div className="col-4 text-center">
          <img
            src={product.img}
            alt={product.name}
            className="img-fluid"
            style={{ maxHeight: "100%", maxWidth: "100%", height: "auto" }}
          />
        </div>
        <div className="col text-center">
          <p>{product.name}</p>
          <hr />
          <div className="row">
            <div className="col">
              <button type="button" className="btn btn-success btn-block">
                Add <FaCartPlus />
              </button>
            </div>
            <div className="col">
              <p style={{ color: "#116CB4" }}>${product.cost}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  products: state.products.items,
  error: state.products.error
});

export default connect(
  mapStateToProps,
  { getProducts }
)(ProductCard);
