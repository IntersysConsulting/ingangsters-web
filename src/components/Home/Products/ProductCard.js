import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { FaCartPlus } from "react-icons/fa";
import "./ProductCard.css";
import { addProductToCart } from "../../Cart/ProductsManager";
import { uploadAndUpdateCart } from "../../../actions/creators/cart";

const ProductCard = ({ product, uploadAndUpdateCart }) => {
  return (
    <div className="container-fluid row-eq-height">
      <div className="row my-1 product-card highlight">
        <div className="col-12 col-sm-12 col-md-4 col-lg-5 text-center vertical-center">
          <Link to={`/details/${product._id}`}>
            <img
              src={product.image}
              alt={product.name}
              className="img-fluid img-resize py-2"
            />
          </Link>
        </div>
        <div className="col-12 col-sm-12 col-md-8 col-lg-7 content-center">
          <Link to={`/details/${product._id}`} className="title-link">
            <div className="product-title">
              <p className="line-clamp title-link">{product.name}</p>
            </div>
          </Link>
          <div className="row">
            <div className="col-lg-3 col-md-12 col-sm-12">
              <p className="price">${product.price}</p>
            </div>
            <div className="col-lg-8 offset-lg-1 col-md-12 col-sm-12">
              <button
                type="button"
                className="btn btn-success btn-block"
                onClick={() => {
                  //WIP: Show animation or alerto to user, "Product added", it can happend in uploadAndUpdateCart
                  addProductToCart(product, product._id, uploadAndUpdateCart);
                }}
              >
                Add <FaCartPlus />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default connect(
  null,
  { uploadAndUpdateCart }
)(ProductCard);
