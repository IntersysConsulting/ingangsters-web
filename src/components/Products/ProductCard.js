import React from "react";
import { Link } from "react-router-dom";
import { FaCartPlus } from "react-icons/fa";
import "./ProductCard.css";

const ProductCard = ({ product }) => {
  const addToCart = () => {
    alert("Added to cart");
  };

  return (
    <div className="container-fluid">
      <div className="row m-1 product-card highlight">
        <div className="col-4 text-center vertical-center">
          <Link to={`/details/${product._id}`}>
            <img
              src={product.image}
              alt={product.name}
              className="img-fluid py-2"
              style={{ maxHeight: "100%", maxWidth: "100%", height: "auto" }}
            />
          </Link>
        </div>
        <div className="col-8 content-center">
          <Link to={`/details/${product._id}`} className="title-link">
            <div className="product-title">
              <p className="line-clamp title-link">{product.name}</p>
            </div>
          </Link>
          <div className="row">
            <div className="col-lg-3 col-md-12 col-sm-12">
              <p className="price">${product.price}</p>
            </div>
            <div className="col-lg-9 col-md-12 col-sm-12">
              <button
                type="button"
                className="btn btn-success btn-block"
                onClick={addToCart}
              >
                Add <FaCartPlus />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ProductCard;
