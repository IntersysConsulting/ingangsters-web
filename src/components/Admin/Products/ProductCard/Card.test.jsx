import React from "react";
import { shallow } from "enzyme";
import ProductCard from "./Card";

describe("ADMIN: Product Card", () => {
  it("renders with random props", () => {
    const component = shallow(
      <ProductCard
        name="LOREM"
        image="https://makeitreal.camp/assets/react-logo-ac85b6b5793992bc49365c389fe88d09b100c758d6981653724ad613764911b2.png"
        stock={123}
        onClick="/random/Path"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
