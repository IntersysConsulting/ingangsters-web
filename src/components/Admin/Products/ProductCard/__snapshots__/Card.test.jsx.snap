// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADMIN: Product Card renders with random props 1`] = `
<Link
  className="productCard"
  to="/random/Path"
>
  <div
    className="productImage"
  >
    <img
      alt="LOREM"
      src="https://makeitreal.camp/assets/react-logo-ac85b6b5793992bc49365c389fe88d09b100c758d6981653724ad613764911b2.png"
    />
  </div>
  <div
    className="productData"
  >
    <p
      className="productName"
    >
      LOREM
    </p>
    <p
      className="productStock"
    >
      Stock: 
      <span
        className="high"
      >
        123
      </span>
    </p>
  </div>
</Link>
`;
